File: tracker/src/main/java/org/matomo/sdk/dispatcher/PacketSender.java
package org.matomo.sdk.dispatcher;


public interface PacketSender {
    /**
     * @return true if successful
     */
    boolean send(Packet packet);

    /**
     * @param timeout in milliseconds
     */
    void setTimeout(long timeout);

    void setGzipData(boolean gzip);
}

File: tracker/src/main/java/org/matomo/sdk/dispatcher/PacketSender.kt
package org.matomo.sdk.dispatcher


interface PacketSender {
    /**
     * @return true if successful
     */
    fun send(packet: Packet): Boolean

    /**
     * @param timeout in milliseconds
     */
    fun setTimeout(timeout: Long)

    fun setGzipData(gzip: Boolean)
}
